ios/AppLovinMAXAdView.mm

```diff
@@ -119,6 +132,392 @@ + (void)destroyNativeUIComponentAdView:(NSNumber *)adViewId
+                .placement = std::string([[event valueForKey: @"placement"] ?: @"" UTF8String]),
+                .revenue = [[event valueForKey: @"revenue"] doubleValue],
+                .revenuePrecision = std::string([[event valueForKey: @"revenuePrecision"] ?: @"" UTF8String]),
+                .latencyMillis = [[event valueForKey: @"latencyMillis"] doubleValue],
+                .dspName = std::string([[event valueForKey: @"dspName"] ?: @"" UTF8String]),
+                .size = {
+                    .width = [[size valueForKey: @"width"] doubleValue],
+                    .height = [[size valueForKey: @"height"] doubleValue],
+                }
+            };
+            
+            adViewEventEmitter->onAdCollapsedEvent(result);
+        }
+    };
+    
+    self.onAdRevenuePaidEvent = [self](NSDictionary *event)
+    {
+        if ( _eventEmitter )
+        {
+            auto adViewEventEmitter = std::static_pointer_cast<AppLovinMAXAdViewEventEmitter const>(_eventEmitter);
+            
+            NSDictionary *size = [event valueForKey: @"size"];
+            
+            AppLovinMAXAdViewEventEmitter::OnAdRevenuePaidEvent result =
+            {
+                .adUnitId = std::string([[event valueForKey: @"adUnitId"] ?: @"" UTF8String]),
+                .adFormat = std::string([[event valueForKey: @"adFormat"] ?: @"" UTF8String]),
+                .adViewId = [[event valueForKey: @"adViewId"] doubleValue],
+                .networkName = std::string([[event valueForKey: @"networkName"] ?: @"" UTF8String]),
+                .networkPlacement = std::string([[event valueForKey: @"networkPlacement"] ?: @"" UTF8String]),
+                .creativeId = std::string([[event valueForKey: @"creativeId"] ?: @"" UTF8String]),
+                .placement = std::string([[event valueForKey: @"placement"] ?: @"" UTF8String]),
+                .revenue = [[event valueForKey: @"revenue"] doubleValue],
+                .revenuePrecision = std::string([[event valueForKey: @"revenuePrecision"] ?: @"" UTF8String]),
+                .latencyMillis = [[event valueForKey: @"latencyMillis"] doubleValue],
+                .dspName = std::string([[event valueForKey: @"dspName"] ?: @"" UTF8String]),
+                .size = {
+                    .width = [[size valueForKey: @"width"] doubleValue],
+                    .height = [[size valueForKey: @"height"] doubleValue],
+                }
+            };
+            
+            adViewEventEmitter->onAdRevenuePaidEvent(result);
+        }
+    };
+}
+
+- (void)updateProps:(Props::Shared const &)props oldProps:(Props::Shared const &)oldProps
+{
+    const auto &oldViewProps = *std::static_pointer_cast<AppLovinMAXAdViewProps const>(_props);
+    const auto &newViewProps = *std::static_pointer_cast<AppLovinMAXAdViewProps const>(props);
+    
+    BOOL isAdUnitIdSet = NO;
+    
+    if ( oldViewProps.adUnitId != newViewProps.adUnitId )
```

Comment:
`isEqualToString:`

Reply:
This `adUnitId` is `std:string`, so the operator should work.